crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -78,6 +101,25 @@ impl<T: ShaderType + WriteInto> StorageBuffer<T> {
     }
 }
 
+/// Stores data to be transferred to the GPU and made accessible to shaders as a dynamic storage buffer.
+///
+/// Dynamic storage buffers can be made available to shaders as some combination of read/write, unlike
+/// [`UniformBuffer`](crate::render_resource::UniformBuffer) which is read-only. Furthermore, dynamic storage buffers
+/// can store much larger data than uniform buffers, which are best suited to relatively small data. Dynamic storage buffers
+/// are particularly well-suited to storing like a Rust-vector (wgpu's "runtime-sized arrays") and have a
+/// [`push`](crate::render_resource::DynamicStorageBuffer::push) method, unlike
+/// [`StorageBuffer`](crate::render_resource::StorageBuffer). Note however that
+/// WebGL2 does not support dynamic storage buffers, so other alternatives to consider are vertex/instance buffers (see
+/// [`BufferVec`](crate::render_resource::BufferVec)), or data textures ([`Texture`](crate::render_resource::Texture)),
+/// depending on what is most appropriate for the use case.
+///
+/// The contained data is stored in system RAM. [`write_buffer`](crate::render_resource::DynamicStorageBuffer::write_buffer)
+/// queues copying of the data from system RAM to VRAM. Dynamic storage buffers must conform to
+/// [std430 alignment/padding requirements]; whenever data is [`push`](crate::render_resource::DynamicStorageBuffer::push)ed
+/// into this structure, it is automatically aligned to these requirements. If data does not need to
+/// be automatically padded or aligned, consider using [`BufferVec`](crate::render_resource::BufferVec).
```

Comment:
I haven't dealt with the line breaks but something like this.
```suggestion
/// queues copying of the data from system RAM to VRAM. The data within a storage buffer binding must conform to
/// [std430 alignment/padding requirements]. `DynamicStorageBuffer` takes care of serialising the inner type to conform to these requirements. Each item [`push`](crate::render_resource::DynamicStorageBuffer::push)ed
/// into this structure will additionally be aligned to meet dynamic offset alignment requirements. If data does not need to
/// be automatically padded or aligned, consider using [`BufferVec`](crate::render_resource::BufferVec).
```